app/controllers/admin/vouchers_controller.rb

```diff
@@ -9,19 +9,33 @@ def new
     end
 
     def create
-      @voucher = Voucher.create(permitted_resource_params.merge(enterprise: @enterprise))
+      case params[:vouchers_flat_rate][:voucher_type]
```

Comment:
If it is `vouchers_flat_rate`, then would the `voucher_type` always be flat rate?

Reply:
I just randomly came back to this and  have to admit confusion. 

If the form is rendered with 
```
    def new
      @voucher = Voucher.new
    end
```
and
```
= form_with model: @voucher, ...
```

Then why are the form fields named after one of the sub-classes `vouchers_flat_rate`?

It seems to work according to the spec so am I missing something?

Reply:
It is a bit confusing indeed, it is because `Vouchers::FlatRate` is the default type on `Voucher#type`, see migration below: 
```ruby
  def up
    # It will set all the vouchers till now to Vouchers::FlatRate
    add_column :vouchers, :type, :string, limit: 255, null: false, default: "Vouchers::FlatRate"
  end
```

So when you do ` @voucher = Voucher.new` you actually create a `Vouchers::FlatRate` hence why the form filed are name `vouchers_flat_rate`

Reply:
Ah ha, thanks for pointing that out. I do remember seeing that last week now..